Remove unnecessary asserts in _correct_storage_aliasing

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #162220
* #162219
* #162218
* __->__ #161694
* #161693
* #161692
* #161634
* #161633
* #161596
* #161595
* #161591
* #161590
* #161586
* #161466

These don't seem necessary to get reasonable errors:
- func was already passed to get_alias_info earlier in the calling function
- all we require for args and outs is the ability to index into them, which will cleanly throw if it fails anyway

Differential Revision: [D81530108](https://our.internmc.facebook.com/intern/diff/D81530108)

swolchok committed:
Remove unnecessary asserts in _correct_storage_aliasing

These don't seem necessary to get reasonable errors:
- func was already passed to get_alias_info earlier in the calling function
- all we require for args and outs is the ability to index into them, which will cleanly throw if it fails anyway

[ghstack-poisoned]

swolchok committed:
rebase on "Remove unnecessary asserts in _correct_storage_aliasing"


These don't seem necessary to get reasonable errors:
- func was already passed to get_alias_info earlier in the calling function
- all we require for args and outs is the ability to index into them, which will cleanly throw if it fails anyway

[ghstack-poisoned]

swolchok commented on torch/utils/_python_dispatch.py:
```diff
@@ -526,9 +526,6 @@ def _correct_storage_aliasing(func, schema_info, args, outs):
     It does this by unsafely overwriting the storage field of the output tensor
     to be the same storage as the input.
     """
-    assert isinstance(func, torch._ops.OpOverload)
```
add back? it was never there. I can add it and see if linters complain though

Skylion007 replied:
@swolchok Yeah, but the typing would have at least been partially enabled for the function scope. This is useful info and we should add it the type def and see if the linters complain.

swolchok committed:
rebase on "Remove unnecessary asserts in _correct_storage_aliasing"


These don't seem necessary to get reasonable errors:
- func was already passed to get_alias_info earlier in the calling function
- all we require for args and outs is the ability to index into them, which will cleanly throw if it fails anyway

[ghstack-poisoned]

swolchok commented:
@swolchok has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81530108).

swolchok committed:
Update on "Remove unnecessary asserts in _correct_storage_aliasing"


These don't seem necessary to get reasonable errors:
- func was already passed to get_alias_info earlier in the calling function
- all we require for args and outs is the ability to index into them, which will cleanly throw if it fails anyway

Differential Revision: [D81530108](https://our.internmc.facebook.com/intern/diff/D81530108)

[ghstack-poisoned]
